cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -0,0 +1,913 @@
+                   LocalNodeConnectInfo (..), connectToLocalNode)
+import           Cardano.Api.Modes (CardanoMode)
+import           Cardano.Api.NetworkId (NetworkId (Mainnet))
+import qualified Cardano.Chain.Genesis
+import qualified Cardano.Chain.UTxO
+import qualified Cardano.Chain.Update
+import qualified Cardano.Crypto
+import qualified Cardano.Crypto.Hash.Blake2b
+import qualified Cardano.Crypto.Hash.Class
+import qualified Cardano.Crypto.Hashing
+import qualified Cardano.Crypto.ProtocolMagic
+import           Cardano.Slotting.Slot (WithOrigin (At, Origin))
+import qualified Cardano.Slotting.Slot as Slot
+import           Network.TypedProtocol.Pipelined (Nat (..))
+import qualified Ouroboros.Consensus.Block.Abstract as Consensus
+import qualified Ouroboros.Consensus.Byron.Ledger.Block as Byron
+import qualified Ouroboros.Consensus.Cardano as Consensus
+import qualified Ouroboros.Consensus.Cardano.Block as Consensus
+import qualified Ouroboros.Consensus.Cardano.CanHardFork as Consensus
+import qualified Ouroboros.Consensus.Config as Consensus
+import qualified Ouroboros.Consensus.HardFork.Combinator.AcrossEras as HFC
+import qualified Ouroboros.Consensus.HardFork.Combinator.Basics as HFC
+import qualified Ouroboros.Consensus.Ledger.Abstract as Ledger
+import qualified Ouroboros.Consensus.Ledger.Extended as Ledger
+import qualified Ouroboros.Consensus.Node.ProtocolInfo as Consensus
+import qualified Ouroboros.Consensus.Shelley.Eras as Shelley
+import qualified Ouroboros.Consensus.Shelley.Ledger.Block as Shelley
+import qualified Ouroboros.Consensus.Shelley.Ledger.Ledger as Shelley
+import qualified Ouroboros.Consensus.Shelley.Protocol as Shelley
+import qualified Ouroboros.Network.Block
+import           Ouroboros.Network.Protocol.ChainSync.ClientPipelined
+                   (ChainSyncClientPipelined (ChainSyncClientPipelined),
+                   ClientPipelinedStIdle (CollectResponse, SendMsgDone, SendMsgRequestNextPipelined),
+                   ClientStNext (..))
+import           Ouroboros.Network.Protocol.ChainSync.PipelineDecision
+import qualified Shelley.Spec.Ledger.BaseTypes as Shelley.Spec
+import qualified Shelley.Spec.Ledger.Credential as Shelley.Spec
+import qualified Shelley.Spec.Ledger.Genesis as Shelley.Spec
+import qualified Shelley.Spec.Ledger.Keys as Shelley.Spec
+import qualified Shelley.Spec.Ledger.PParams as Shelley.Spec
+
+-- | Get the environment and initial ledger state.
+initialLedgerState
+  :: FilePath
+  -- ^ Path to the cardano-node config file (e.g. <path to cardano-node project>/configuration/cardano/mainnet-config.json)
+  -> IO (Env, LedgerState)
+  -- ^ The environment and initial ledger state
+initialLedgerState dbSyncConfFilePath = do
+  dbSyncConf <- readDbSyncNodeConfig (NodeConfigFile dbSyncConfFilePath)
+  genConf <- fmap (either (error . Text.unpack . renderDbSyncNodeError) id) $ runExceptT (readCardanoGenesisConfig dbSyncConf)
```

Comment:
If this is eventually intended for `cardano-cli` let's return an `Either` so we can handle errors there.

Reply:
Yes... actually I used `error` in a few places and need to turn these into propper errors

Reply:
@Jimbo4350 I'm using `ExceptT` now. Is that ok with you?